### PR TITLE
[ioctl] Build did deregister command line into new ioctl

### LINKED ISSUE #3667: [ioctl] build did deregister command line into new ioctl
### What would you like to be added
 - Step 1: create `didderegister.go` and `didderegister_test.go` files in `./ioctl/newcmd/did`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didderegister.go in `didderegister.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didderegister.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `didderegister` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3667  

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewDeregisterCmd

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
****

### CODE REVIEW

File: ioctl/newcmd/did/didderegister_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/golang/mock/gomock"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi/mock_iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/iotex-address/address"
```
Comment (huof6829) [inline ioctl/newcmd/did/didderegister_test.go:19]: move line19 to  line17  group

File: ioctl/newcmd/did/didderegister.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+// Multi-language support
+var (
+	_deregisterCmdUses = map[config.Language]string{
+		config.English: "deregister (CONTRACT_ADDRESS|ALIAS)",
+		config.Chinese: "deregister (合约地址|别名)",
+	}
+	_deregisterCmdShorts = map[config.Language]string{
+		config.English: "Deregister DID on IoTeX blockchain",
+		config.Chinese: "Deregister 在IoTeX链上注销DID",
```
[the remark this answers is not in the document]
Comment (LuckyPigeon) [reply ioctl/newcmd/did/didderegister.go:28]: There are some same patterns of line in `didgethash`, `didgeturi` and `didregister`, I will modify them accordingly as well.

[the remark this answers is not in the document]
Comment (huof6829) [reply ioctl/newcmd/did/didderegister.go]: I modified in #3668

File: ioctl/newcmd/did/didderegister.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+// Multi-language support
+var (
+	_deregisterCmdUses = map[config.Language]string{
+		config.English: "deregister (CONTRACT_ADDRESS|ALIAS)",
+		config.Chinese: "deregister (合约地址|别名)",
+	}
+	_deregisterCmdShorts = map[config.Language]string{
+		config.English: "Deregister DID on IoTeX blockchain",
+		config.Chinese: "在IoTeX链上注销DID",
```
Comment (huof6829) [inline ioctl/newcmd/did/didderegister.go:28]: add flag shorts.  also in did register 
https://github.com/iotexproject/iotex-core/blob/ee04a051811a9c23b33a291fb66c2de957b78bf3/ioctl/newcmd/action/actiontransfer.go#L29

**Review by huof6829 — APPROVED** [review]

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**